scripts/undemultiplexed_index.py

```diff
@@ -51,165 +41,208 @@
-            ID = cont_name
-        try:
-            self.run_id = seq_run[0].udf['Run ID']
-        except:
-            logging.info('could not get Run ID')
-        logging.info('looking for sequencing setup')
-        try:
-            Read_1_Cycles = seq_run[0].udf['Read 1 Cycles']
+    
+    def _get_run(self, cont_name):
+        """Getting parrent sequencing process and process type"""
+        miseq = lims.get_processes(udf = {'Reagent Cartridge ID' : cont_name},
+                                type = 'MiSeq Run (MiSeq) 4.0')
+        hiseq = lims.get_processes(udf = {'Flow Cell ID' : cont_name},
+                                type = 'Illumina Sequencing (Illumina SBS) 4.0')
+        hiseq_X10 = lims.get_processes(udf = {'Flow Cell ID' : cont_name},
+                                type = 'Illumina Sequencing (HiSeq X) 1.0')
+        if miseq:
+            self.seq_run = miseq[0]
+            self.run_type = 'MiSeq'
+        elif hiseq:
+            self.seq_run = hiseq[0]
             try:
-                Read_2_Cycles = seq_run[0].udf['Read 2 Cycles']
-                self.single = False
+                self.run_type = self.seq_run.udf['Flow Cell Version']
             except:
-                self.single = True
-        except:
-            sys.exit("Could not get sequencing set up. Make sure 'Read 1 Cycles' (and 'Read 2 Cycles' if applicable) is set in the sequencing step.")
+                sys.exit("Missing field 'Flow Cell Version' in sequencing process")
+        elif hiseq_X10:
+            self.seq_run = hiseq_X10[0]
+            self.run_type = 'HiSeqX10'
+        else:
+            sys.exit("run not found")
+
+    def _get_run_id(self):
+        if self.run_udfs.has_key('Run ID'):
+            self.process.udf['Run ID'] = self.run_udfs['Run ID']
+            set_field(self.process)
+
+    def _get_cycles(self):
+        """To find out if it was a single end or paired end run."""
+        if self.run_udfs.has_key('Read 1 Cycles'):
+            self.read_length = self.run_udfs['Read 1 Cycles']
+        else:
+            sys.exit("Could not get 'Read 1 Cycles' from the sequencing step.")
+        if self.run_udfs.has_key('Read 2 Cycles'):
+            self.single = False
+
+    def _get_file_path(self, cont_name):
+        """File path for HiSeq, Miseq and Xten"""
+        if self.run_type == 'MiSeq':
+            path_id = self.run_udfs['Flow Cell ID']
+            data_folder = 'miseq_data'
+        elif self.run_type == 'HiSeqX10':
+            path_id = cont_name
+            data_folder = 'HiSeq_X_data'
+        else:
+            data_folder = 'hiseq_data'
+            path_id = cont_name
         try:
-            return glob.glob(("/srv/mfs/*iseq_data/*{0}/Unaligned/Basecall_Stats_*/".format(ID)))[0]
+            self.file_path = glob.glob(("/srv/mfs/{0}/*{1}/Unaligned/"
+                           "Basecall_Stats_*/".format(data_folder, path_id)))[0]
```

Comment:
@guillermo-carrasco what about HiSeqX, we save in the same way on mfs these files?